pkg/pillar/cmd/tpmmgr/tpmmgr.go

```diff
@@ -945,18 +1050,36 @@ func Run(ps *pubsub.PubSub) {
 			//FIXME: On some platforms, these key creations are failing. Till we root-cause,
 			//use Errorf instead of Fatalf. The features that use these keys are not enabled yet.
 			if err = createKey(TpmEKHdl, tpm2.HandleEndorsement, defaultEkTemplate, false); err != nil {
-				log.Errorf("Error in creating Endorsement key: %v ", err)
+				//are we rebooting? capture that info for EPIPE errors
+				deviceReboot, ok := readNodeAgentStatus(&ctx)
```

Comment:
Is there a reason why a future (in 30 seconds or so) reboot would be relevant to these TPM operations failing vs. succeeding?

Reply:
In some cases, EPIPE is due to the ongoing reboot.  I could manually repro that case, but in some setups, we could not repro this EPIPE Fatal. In case we see this EPIPE Fatal again in those setups, these logs will help confirm that the EPIPE is due to the ongoing reboot. (If that is the case, we might have to avoid  doing a Fatal in such cases, that will do once we confirm if that is the case)

Reply:
Would it make sense to add a check in nodeagent (where we check for running App Instances aka DomainStatus) to also add a check whether any tpmmgr operations are in progress? No need to do this as part of this PR though.

Reply:
Actually this issue is more pronounced when there is a long-lived fd to the TPM char device. To mitigate it we can do 2 things:
a) nodeagent publishes about-to-reboot notification
b) tpmmgr can check for reboot-in-progress and then refuse any request to TPM device, and return EAGAIN (?) to the caller
c) nodeagent can wait for tpmmgr to publish a message that there is no pending operations
d) nodeagent after receiving this notification can go ahead with the reboot

We can tackle this in a different PR?

Reply:
A separate PR makes sense. FWIW I think we already have a) in NodeAgentStatus